### PR TITLE
feat: update comment

### PR DESCRIPTION
**Pull request by idowuseyi**
**What does the PR do?**
This PR introduces new API endpoints to update comments. It includes server-side logic for handling comment update, along with appropriate validation and error handling.

How should this be manually tested?

To Update a Comment:

Send a PATCH request to /api/v1/comments/update/:comment_id with a valid UpdateCommentDto payload and a valid comment_id in the request.
Example:

`PATCH /api/v1/comments/update/:comment_id`

{
  "content": "This is the update to the original comment"
}

Successful Response

On a successful comment update, the API should return a 200 SUCCESS status code.

```json
{
      "status": "status_code",
      "message": "Comment updated successfully",
      "data": {
            "id": "84488317-234b-41b5-a2d1-07d47a50be8e",
            "created_at": "2024-08-08T06:27:40.309Z",
            "updated_at": "2024-08-08T06:59:59.030Z",
            "comment": "I just update this comment now",
            "model_id": "1234",
            "model_type": "post"
            }
      "commentedBy" : "John Smith"
}
```

Error Handling:

When the current user is not authourized:

```json
{
          "status": "error",
          "message": "User not authorized",
          "status_code": "401",
 }
```

When the current user is not found:

```json
{
          "status": "error",
          "message": "User nof found",
          "status_code": "404",
 }
```

Error During coment update

If the comment creation is not successful:

```json
{
  "status": "Error",
  "message": "Error creating comment",
  "status_code": 500
}
```

 **Checklist of what has been done**
- [X ] Define API endpoint to update comment.
- [X ]  Validate and handle authentication token.
- [ X]  Implement logic to update a comment.
- [ X]  Handle success and error responses.
- [ X]  Test Successful response

**API Test Screenshot**
Screenshot from 2024-08-08 07-59-37

** Jest Test Screenshot **
Screenshot from 2024-08-07 15-46-24

Referenced Linear Ticket: [Ticket-Link](https://linear.app/team-bingo/issue/BAC2-134/endpoint-to-edit-comment-or-212)
Referenced Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/212)

### CODE REVIEW

File: src/modules/comments/comments.controller.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export class CommentsController {
+  constructor(private readonly commentsService: CommentsService) {}
+  @Post('add')
+  async addComment(@Body() createCommentDto: CreateCommentDto, @Request() req): Promise<CommentResponseDto> {
+    const { userId } = req.user;
+    return await this.commentsService.addComment(createCommentDto, userId);
+  }
+
+  @ApiOperation({ summary: 'Update a comment' })
+  @ApiResponse({ status: 200, description: 'The comment has been successfully updated.' })
```
Comment (Cobidex) [inline src/modules/comments/comments.controller.ts:20]: ensure proper swagger documentation of response and request body

File: src/modules/comments/comments.controller.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  @Patch('update/:id')
+  async updateComment(
+    @Param('id') id: string,
```
Comment (Cobidex) [inline src/modules/comments/comments.controller.ts:23]: use validation pipe to ensure only uuid string gets through

File: src/modules/comments/comments.controller.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @Body() updateCommentDto: UpdateCommentDto,
+    @Request() req
+  ): Promise<CommentResponseDto> {
+    const { userId } = req.user;
+    return await this.commentsService.updateComment(id, userId, updateCommentDto);
```
Comment (Cobidex) [inline src/modules/comments/comments.controller.ts:28]: remove await

File: src/modules/comments/comments.service.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  async addComment(createCommentDto: CreateCommentDto, userId: string): Promise<CommentResponseDto> {
+    const { model_id, model_type, comment } = createCommentDto;
+
+    if (!comment || comment.trim().length === 0) {
+      throw new BadRequestException('Comment cannot be empty');
+    }
+
+    const user = await this.userRepository.findOne({ where: { id: userId } });
+    if (!user) {
+      throw new NotFoundException('User not found');
```
Comment (Cobidex) [inline src/modules/comments/comments.service.ts:27]: use custom exception filters

File: src/modules/comments/comments.service.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        comment,
+      });
+
+      const loadComment = await this.commentRepository.save(Comment);
+      return {
+        message: 'Comment added successfully!',
+        savedComment: loadComment,
+        commentedBy,
+      };
+    } catch (error) {
```
Comment (Cobidex) [inline src/modules/comments/comments.service.ts:45]: remove try catch

File: src/modules/comments/comments.controller.ts (hunk 1 above)
Comment (idowuseyi) [reply src/modules/comments/comments.controller.ts:20]: Fixed in another waiting PR

File: src/modules/comments/comments.controller.ts (hunk 2 above)
Comment (idowuseyi) [reply src/modules/comments/comments.controller.ts:23]: Fixed

File: src/modules/comments/comments.service.ts (hunk 5 above)
Comment (idowuseyi) [reply src/modules/comments/comments.service.ts:45]: fixed

**Pull request closed without merging**